lib/ui/screens/home/home_view.dart

```diff
@@ -28,32 +28,39 @@ class HomeView extends StatelessWidget {
         onPressed: controller.openSearch,
       );
 
-  Widget _buildContent() => CustomScrollView(
-        slivers: <Widget>[
-          SliverPadding(
-            padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 32),
-            sliver: SliverList.separated(
-              itemBuilder: (_, int index) {
-                final HomeModelPost post = model.posts[index];
-                return PostPreview(
-                  onPostTap: (String postId) =>
-                      controller.openPost(postId: postId),
-                  title: post.title,
-                  content: post.body,
-                  account: PostPreviewAccount(
-                    name: 'Account Name',
-                    titles: <String>['Title 1', 'Title 2'],
-                    imageUrl: Uri.parse(
-                      'https://images.unsplash.com/photo-1534528741775-53994a69daeb?q=80&w=3276&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
-                    ),
-                  ),
-                  postId: post.postId,
-                );
-              },
-              separatorBuilder: (_, __) => const SizedBox(height: 16),
-              itemCount: model.posts.length,
-            ),
+  Widget _buildContent() => Scaffold(
+        body: model.when(
+          data: (List<HomeModelPost> posts) => CustomScrollView(
```

Comment:
**suggestion:** extract each case into separate method to clean up the page structure. 
Apply everywhere in this PR.